src/main/java/de/tum/in/www1/artemis/service/StudentExamService.java

```diff
@@ -248,6 +260,62 @@ else if (exercise instanceof ModelingExercise) {
+     * The assessment is counted as {@link AssessmentType#SEMI_AUTOMATIC} to make sure it is not considered for manual assessment,
+     * see {@link StudentParticipationRepository#findByExerciseIdWithLatestSubmissionWithoutManualResultsAndNoTestRunParticipation}.
+     * The assessor is artemis_admin.
+     *
+     * @param examId the exam id
+     * @return returns the number of assessedSubmissions
+     */
+    public int assessUnsubmittedStudentExams(final Long examId) {
+        int numberOfAssessedSubmissions = 0;
+        User artemisAdmin = userService.getUserWithGroupsAndAuthorities("artemis_admin");
+        Set<StudentExam> unsubmittedStudentExams = findAllUnsubmittedStudentExams(examId);
+        Map<User, List<Exercise>> exercisesOfUser = unsubmittedStudentExams.stream().collect(Collectors.toMap(StudentExam::getUser, studentExam -> studentExam.getExercises()
+                .stream().filter(exercise -> exercise instanceof ModelingExercise || exercise instanceof TextExercise).collect(Collectors.toList())));
+        for (final var user : exercisesOfUser.keySet()) {
+            final var studentParticipations = participationService.findByStudentIdAndIndividualExercisesWithEagerSubmissionsResult(user.getId(), exercisesOfUser.get(user));
+            for (final var studentParticipation : studentParticipations) {
+                if (studentParticipation.findLatestSubmission().isPresent() && studentParticipation.findLatestSubmission().get().getLatestResult() == null) {
+                    // get last submission
+                    final var latestSubmission = studentParticipation.findLatestSubmission().get();
+
+                    // create result with 0 points
+                    Result result = new Result();
+                    result.setParticipation(studentParticipation);
+                    result.setAssessor(artemisAdmin);
+                    result.setCompletionDate(ZonedDateTime.now());
+                    result.setScore(0L);
+                    result.rated(true);
+                    result.setAssessmentType(AssessmentType.SEMI_AUTOMATIC);
+                    result = submissionService.saveNewResult(latestSubmission, result);
+
+                    var feedback = new Feedback();
+                    feedback.setCredits(0.0);
+                    feedback.setDetailText("You did not submit your exam");
+                    feedback.setPositive(false);
+                    feedback.setText("You did not submit your exam");
+                    feedback.setType(FeedbackType.AUTOMATIC);
+                    feedback = feedbackRepository.save(feedback);
+                    feedback.setResult(result);
+                    result.setFeedbacks(List.of(feedback));
+                    resultRepository.save(result);
+                    numberOfAssessedSubmissions++;
+                }
+            }
+        }
+        return numberOfAssessedSubmissions;
+    }
+
+    public int assessEmptySubmissions(final Long examId) {
```

Comment:
What's the point of this method? It fetches an instructor and student exams but it doesn't really use them right?

Reply:
I guess the implementation is not finished yet

Reply:
Yeah I had forgotten to remove the ready for review tag. I pushed the implementation now